src/Microsoft.IdentityModel.Tokens/Validators.cs

```diff
@@ -93,7 +93,8 @@ public static void ValidateAudience(IEnumerable<string> audiences, SecurityToken
             if (string.IsNullOrWhiteSpace(validationParameters.ValidAudience) && (validationParameters.ValidAudiences == null))
                 throw LogHelper.LogExceptionMessage(new SecurityTokenInvalidAudienceException(LogMessages.IDX10208) { InvalidAudience = Utility.SerializeAsSingleCommaDelimitedString(audiences) });
 
-            if (!audiences.Any())
+            // If audiences has no elements or all elements are null or empty, throw
+            if (!audiences.Any(s => !s.IsNullOrEmpty()))
```

Comment:
I made this check a bit more strict since it is more likely we will deal with multiple audiences. Is it reasonable to do so?

Reply:
We shouldn't change the behavior if it's not strictly necessary. Thoughts @brentschmaltz  and @keegan-caruso

Reply:
This sounds right. Lets not change this.